packages/multistore/src/types.ts

```diff
@@ -3,19 +3,24 @@
  */
 export type StoreConfig = Record<string, any>;
 
+/**
+ * Utility for typing functions which can be defined as either async or sync.
+ */
+type MaybePromise<T> = T | Promise<T>;
```

Comment:
Isn't it better to use `Awaited`?

Reply:
So far I've been using `Awaited` only to extract the return type of async functions, not to type them. I am not sure how does `Awaited` fit in here. Could you show a full example?